File: delete/poo-mini/python/vehiculos/vehiculo.py
from abc import ABC, abstractmethod

class Vehiculo(ABC):
    def __init__(self, marca: str):
        self._marca = marca          # convención "protegido"
        self._velocidad = 0

    @property
    def marca(self) -> str:
        return self._marca

    @property
    def velocidad(self) -> int:
        return self._velocidad

    def acelerar(self, delta: int) -> None:
        if delta < 0:
            raise ValueError("delta debe ser >= 0")
        self._velocidad += delta

    def frenar(self, delta: int) -> None:
        if delta < 0:
            raise ValueError("delta debe ser >= 0")
        self._velocidad = max(0, self._velocidad - delta)

    @abstractmethod
    def mover(self) -> str:
        ...

File: delete/poo-mini/java/src/main/java/com/poo/vehiculos/Vehiculo.java
package com.poo.vehiculos;

public abstract class Vehiculo {
    private final String marca;
    private int velocidad;

    public Vehiculo(String marca) {
        this.marca = marca;
        this.velocidad = 0;
    }
    public String getMarca() { return marca; }
    public int getVelocidad() { return velocidad; }

    public void acelerar(int delta) {
        if (delta < 0) throw new IllegalArgumentException("delta debe ser >= 0");
        this.velocidad += delta;
    }
    public void frenar(int delta) {
        if (delta < 0) throw new IllegalArgumentException("delta debe ser >= 0");
        this.velocidad = Math.max(0, this.velocidad - delta);
    }

    public abstract String mover();
}
